Source/WebKit/WebProcess/cocoa/RemoteCaptureSampleManager.cpp

```diff
@@ -229,7 +234,31 @@ void RemoteCaptureSampleManager::RemoteAudio::startThread()
             m_source->remoteAudioSamplesAvailable(currentTime, *m_buffer, *m_description, m_frameChunkSize);
         } while (!m_shouldStopThread);
     };
-    m_thread = Thread::create("RemoteCaptureSampleManager::RemoteAudio thread"_s, WTFMove(threadLoop), ThreadType::Audio, Thread::QOS::UserInteractive);
+
+    // Create a thread with Realtime scheduling, so that this thread does not become deprioritized
+    // during heavy loads. Realtime threads have a runtime cost, so ensure no more than two realtime
+    // capture threads can be created simultaneously.
+    // FIXME: Coalesce these threads to allow a single realtime thread to service every RemoteAudio instance.
+    bool shouldCreateRealtimeThread = s_realtimeThreadCount < s_maximumConcurrentRealtimeThreads;
```

Comment:
this seems racy, doesn't it matter?

Reply:
It probably doesn't, but I can update this to an atomic compare and increment.

Reply:
Can we add `ASSERT(isMainRunLoop())` for now?